neqo-common/src/codec.rs

```diff
@@ -15,6 +15,22 @@ pub struct Decoder<'a> {
 }
 
 impl<'a> Decoder<'a> {
+    /// Return the minimal encoded length of a varint with value `v`.
+    ///
+    /// # Panics
+    ///
+    /// When `v` is too large.
+    #[must_use]
+    pub const fn minimal_varint_len(v: u64) -> usize {
+        match () {
+            () if v < (1 << 6) => 1,
+            () if v < (1 << 14) => 2,
+            () if v < (1 << 30) => 4,
+            () if v < (1 << 62) => 8,
+            () => panic!("Varint value too large"),
+        }
+    }
+
```

Comment:
Perhaps I wasn't clear.  This code duplicates the code in the encoder, so you can just use the version in the encoder rather than duplicating it.

```suggestion
```

I'm open to suggestions about a singular place.  You might also do this:

```suggestion
    /// Return the minimal encoded length of a varint with value `v`.
    ///
    /// # Panics
    ///
    /// When `v` is too large.
    #[must_use]
    pub const fn minimal_varint_len(v: u64) -> usize {
        Encoder::varint_len(v)
    }
```

Reply:
Ah! Yeah, sorry that I misunderstood.